pyquil/quilatom.py

```diff
@@ -153,51 +161,54 @@ class Label(QuilAtom):
     :param string label_name: The label name.
     """
 
-    def __init__(self, label_name):
+    def __init__(self, label_name: str) -> None:
         self.name = label_name
 
-    def out(self):
+    def out(self) -> str:
         return "@{name}".format(name=self.name)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return "@{name}".format(name=self.name)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "<Label {0}>".format(repr(self.name))
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         return isinstance(other, Label) and other.name == self.name
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(self.name)
 
 
 class LabelPlaceholder(QuilAtom):
-    def __init__(self, prefix="L"):
+    def __init__(self, prefix: str = "L") -> None:
         self.prefix = prefix
 
     def out(self):
         raise RuntimeError("Label has not been assigned a name")
 
-    def __str__(self):
+    def __str__(self) -> str:
         return repr(self)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "<LabelPlaceholder {} {}>".format(self.prefix, id(self))
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         return isinstance(other, LabelPlaceholder) and id(other) == id(self)
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(id(self))
 
 
-def format_parameter(element):
+ParameterDesignator = Union['Expression', 'MemoryReference', np.int_, int, float, complex]
```

Comment:
Should `complex` be included here? `format_parameter` supports `complex` values, and pyquil doesn't complain if I pass a complex value as a gate parameter, but quilc will choke on it. See e.g. the discussion in https://github.com/rigetti/quilc/issues/313

Reply:
If `quilc` says no, then pyQuil should say no.

Reply:
Perhaps file an issue (1000????)

Reply:
Filed: #1009